lib/wovnrb/html_replacers/input_replacer.rb

```diff
@@ -5,23 +5,33 @@ def initialize(text_index)
     end
 
     def replace(dom, lang)
-      dom.xpath('//input').select { |node|
+      dom.xpath('//input').each do |node|
         next if wovn_ignore?(node)
-        (node.get_attribute('value') &&
-        node.get_attribute('value') != '' &&
-        node.get_attribute('type') &&
-        node.get_attribute('type') != 'text' &&
-        node.get_attribute('type') != 'hidden' &&
-        node.get_attribute('type') != 'password' &&
-        node.get_attribute('type') != 'url' &&
-        node.get_attribute('type') != 'search')
-      }.each do |node|
-        node_value = node.get_attribute('value').strip
-        # shouldn't need size check, but for now...
-        if @text_index[node_value] && @text_index[node_value][lang.lang_code] && @text_index[node_value][lang.lang_code].size > 0
-          node.set_attribute('value', replace_text(node.get_attribute('value'), @text_index[node_value][lang.lang_code][0]['data']))
-        end
+
+        set_attribute('value', node, lang)       if replaceable_value? node
+        set_attribute('placeholder', node, lang) if replaceable_placeholder? node
+      end
+    end
+
+    private
+
+    def set_attribute(name, node, lang)
+      node_value = node.get_attribute(name).strip
+      if @text_index[node_value] && @text_index[node_value][lang.lang_code] && @text_index[node_value][lang.lang_code].size > 0
+        node.set_attribute(name, replace_text(node_value, @text_index[node_value][lang.lang_code][0]['data']))
       end
     end
+
+    def replaceable_value?(node)
+      return false unless node.get_attribute('type') == 'submit'
```

Comment:
I supported reset type  https://github.com/WOVNio/wovnrb/pull/83/commits/faeaec4c24cf423f8c5616077242be98ab29c35b

Value of radio type should be translated. Because, It is used  in server.